test/parallel/test-debug-brk.js

```diff
@@ -1,35 +1,71 @@
 'use strict';
 
 const common = require('../common');
+const assert = require('assert');
 const spawn = require('child_process').spawn;
 
-var procStderr = '';
-var agentStdout = '';
-var needToSpawnAgent = true;
-var needToExit = true;
-
-const procArgs = [`--debug-brk=${common.PORT}`, '-e', '0'];
-const proc = spawn(process.execPath, procArgs);
-proc.stderr.setEncoding('utf8');
-
-const exitAll = common.mustCall((processes) => {
-  processes.forEach((myProcess) => { myProcess.kill(); });
-});
-
-proc.stderr.on('data', (chunk) => {
-  procStderr += chunk;
-  if (/Debugger listening on/.test(procStderr) && needToSpawnAgent) {
-    needToSpawnAgent = false;
-    const agentArgs = ['debug', `localhost:${common.PORT}`];
-    const agent = spawn(process.execPath, agentArgs);
-    agent.stdout.setEncoding('utf8');
-
-    agent.stdout.on('data', (chunk) => {
-      agentStdout += chunk;
-      if (/connecting to .+ ok/.test(agentStdout) && needToExit) {
-        needToExit = false;
-        exitAll([proc, agent]);
+let run = () => {};
+function test(extraArgs, stdoutPattern) {
+  const next = run;
+  run = () => {
+    var procStdout = '';
+    var procStderr = '';
+    var agentStdout = '';
+    var debuggerListening = false;
+    var outputMatched = false;
+    var needToSpawnAgent = true;
+    var needToExit = true;
+
+    const procArgs = [`--debug-brk=${common.PORT}`].concat(extraArgs);
+    const proc = spawn(process.execPath, procArgs);
+    proc.stderr.setEncoding('utf8');
+
+    const tryStartAgent = () => {
+      if (debuggerListening && outputMatched && needToSpawnAgent) {
+        needToSpawnAgent = false;
+        const agentArgs = ['debug', `localhost:${common.PORT}`];
+        const agent = spawn(process.execPath, agentArgs);
+        agent.stdout.setEncoding('utf8');
+
+        agent.stdout.on('data', (chunk) => {
+          agentStdout += chunk;
+          if (/connecting to .+ ok/.test(agentStdout) && needToExit) {
+            needToExit = false;
+            exitAll([proc, agent]);
+          }
+        });
       }
+    };
+
+    const exitAll = common.mustCall((processes) => {
+      processes.forEach((myProcess) => { myProcess.kill(); });
     });
-  }
-});
+
+    if (stdoutPattern != null) {
+      proc.stdout.on('data', (chunk) => {
+        procStdout += chunk;
+        outputMatched = outputMatched || stdoutPattern.test(procStdout);
+        tryStartAgent();
+      });
+    } else {
+      outputMatched = true;
+    }
+
+    proc.stderr.on('data', (chunk) => {
+      procStderr += chunk;
+      debuggerListening = debuggerListening ||
+        /Debugger listening on/.test(procStderr);
```

Comment:
Tiny style nit: line continuations should have four spaces of indent.  The linter is supposed to check that but it's possible it gets thrown off by the binary operator.